cpp/src/strings/search/find.cu

```diff
@@ -407,6 +454,69 @@ std::unique_ptr<column> contains_warp_parallel(strings_column_view const& input,
   return results;
 }
 
+std::vector<std::unique_ptr<column>> multi_contains_warp_parallel(
+  strings_column_view const& input,
+  std::vector<std::reference_wrapper<string_scalar>> const& targets,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr)
+{
+  auto const num_targets = static_cast<size_type>(targets.size());
+  CUDF_EXPECTS(not targets.empty(), "Must specify at least one target string.");
+  CUDF_EXPECTS(std::all_of(targets.begin(),
```

Comment:
We end up with repeated sync because of 1 byte pageable transfers for `is_valid` call. This doesn't seem to be a bottleneck for low number of targets but could become one if we were searching for more targets. I think we could avoid this if targets were in a column instead since the null mask is a contiguous allocation
![image](https://github.com/rapidsai/cudf/assets/23545205/bded3a6b-fb3a-417e-a575-eef19ad49131)

Reply:
Fair point (although the `contains_warp_parallel` kernel might have the same challenge).  I'll hit this on the next go around.

Reply:
https://github.com/rapidsai/cudf/pull/15536#discussion_r1566415863